sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -51,6 +51,11 @@ case class CostBasedJoinReorder(conf: CatalystConf) extends Rule[LogicalPlan] wi
 
   def reorder(plan: LogicalPlan, output: AttributeSet): LogicalPlan = {
     val (items, conditions) = extractInnerJoins(plan)
+    // Find the star schema joins. Currently, it returns the star join with the largest
+    // fact table. In the future, it can return more than one star join (e.g. F1-D1-D2
+    // and F2-D3-D4).
+    val starJoinPlans = StarSchemaDetection(conf).findStarJoins(items, conditions.toSeq)
```

Comment:
The method splitting is good, let's remove this `val starJoinPlans` since it's not used.
In fact I think we could first do something like this in `CostBasedJoinReorder`:
```
if (conf.cboEnabled && conf.joinReorderEnabled) {
  do join reordering
} else if (conf.starSchemaJoinReorder) {
  use star schema
} else {
  plan
}
```
But since the current algorithm is in the `Operator Optimizations` batch, the tree structure is different from that in `CostBasedJoinReorder`, will moving into `CostBasedJoinReorder` break some logic in the algorithm?
Other than that I'm fine with the current structure.